### ISSUE TITLE
Cannot enter whitespace with trim="blur" (ngx-trim-directive)

### ISSUE THREAD

**Issue by tpeter1985** [KingMario/packages#13]
In our project we have some fields where we cannot insert or append whitespaces.
We had no problems with 1.0.3 but we had to update angular and the _ngx-trim-directive_ package (from 1.03 to 1.10).

From my analysis I think the problem is:
If you have multiple fields at once that bind to the same value and they all use _trim="blur"_ a change in one field causes the call of _writeValue()_ in the _valueAccessor_.

You can view a sample of the described behaviour here: https://stackblitz.com/edit/ngx-trim-directive-issue-13

What are your reasons for doing a trim in _writeValue()_?

In our case we don't need to trim the values initially and we don't set the values programmatically, so a solution for our scenario I can imagine would be: Let users of the package turn off trimming in _writeValue()_ optionally.

What do you think?
I would create a Pull Request if this would be a solution and you want it.
But I want to know what you think before starting coding something.

**Comment by KingMario** [comment]
Hi @tpeter1985,

I think an option to turn off the trimming in _writeValue()_ is good. Please create a Pull Request to add it. And if it doesn't cost you too much time, please also add test cases for it.

Thanks.

**Comment by KingMario** [comment]
@tpeter1985
The reason for doing a trimming in _writeValue()_ is to trim the value initially and trim the value while it's updated programmatically. I think it's a feature that might be required by users of the directive.

**Comment by tpeter1985** [comment]
@KingMario 
Sorry for my late reply but I got sick last week.
I created a pull request with the possibility to turn off the _trim()_ in _writeValue()_ by setting _trimOnWriteValue_ input property to _false_. The default value of the input property is true so the default behaviour does not change for others.
In our case it's valid to turn off auto-trimming because we do not update the value programmatically and we do not need initial auto-trimming functionality.

I would appreciate seeing this feature in your package.
Thanks
